### PR TITLE
feat(registry): add rustfmt

### PR DESCRIPTION
**Pull request by jayp0521**

### CODE REVIEW

**Comment by jayp0521** [comment]
https://github.com/williamboman/mason.nvim/issues/605

File: lua/mason-registry/rustfmt/init.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            return _.compose(_.format(template_string), _.gsub("^v", ""))
+        end
+
+        github
+            .untargz_release_file({
+                repo = "rust-lang/rustfmt",
+                asset_file = coalesce(
+                    when(platform.is.mac_x64, release_file "rustfmt_macos-x86_64_%s.tar.gz"),
+                    when(platform.is.linux_x64, release_file "rustfmt_linux-x86_64_%s.tar.gz"),
+                    when(platform.is.win_x64, release_file "rustfmt_windows-x86_64-gnu_%s.tar.gz")
```
Comment (jayp0521) [inline lua/mason-registry/rustfmt/init.lua:26]: i don't know much about windows, https://github.com/rust-lang/rustfmt/releases

Comment (williamboman) [inline lua/mason-registry/rustfmt/init.lua:26]:
```suggestion
                    when(platform.is.win_x64, release_file "rustfmt_windows-x86_64-msvc_%s.tar.gz")
```

Comment (williamboman) [reply lua/mason-registry/rustfmt/init.lua:26]: GNU on Windows require Cygwin, Mingw, or et al. to be installed - in Mason the `win` target is for "Windows native" platforms (i.e. `msvc`)

**Comment by jayp0521** [comment]
good to go?

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/lua/mason-registry/rustfmt/init.lua b/lua/mason-registry/rustfmt/init.lua
index d838eb28..e8a03daf 100644
--- a/lua/mason-registry/rustfmt/init.lua
+++ b/lua/mason-registry/rustfmt/init.lua
@@ -21,7 +21,7 @@ return Pkg.new {
             .untargz_release_file({
                 repo = "rust-lang/rustfmt",
                 asset_file = coalesce(
-                    when(platform.is.mac_x64, release_file "rustfmt_macos-x86_64_%s.tar.gz"),
+                    when(platform.is.mac, release_file "rustfmt_macos-x86_64_%s.tar.gz"),
                     when(platform.is.linux_x64, release_file "rustfmt_linux-x86_64_%s.tar.gz"),
                     when(platform.is.win_x64, release_file "rustfmt_windows-x86_64-msvc_%s.tar.gz")
                 ),
```

**Comment by jinzhongjia** [comment]
rustfmt download errors, not found, the v in the download address was replaced with a space, 
 https://github.com/williamboman/mason.nvim/blob/f5b1a206c4335e1ccfd4f68feef63ba8950e3e90/lua/mason-registry/rustfmt/init.lua#L17
after remove ` _.gsub("^v", "")`, download works fine, but  console Link `target "/home/jin/.local/share/nvim/mason/packages/rustfmt/rustfmt" does not exist.`
I don't know where the downloaded files go？
图片

**Comment by alourie** [comment]
> rustfmt download errors, not found, the v in the download address was replaced with a space,
> 
> https://github.com/williamboman/mason.nvim/blob/f5b1a206c4335e1ccfd4f68feef63ba8950e3e90/lua/mason-registry/rustfmt/init.lua#L17
> 
> after remove ` _.gsub("^v", "")`, download works fine, but console print `Link target "/home/jin/.local/share/nvim/mason/packages/rustfmt/rustfmt" does not exist.`
> I don't know where the downloaded files go？
> 图片

I'm having the same issues (both with wrong removal of `v` in the version as well as the linker error)

**Comment by jinzhongjia** [comment]
> > rustfmt download errors, not found, the v in the download address was replaced with a space,
> > 
> > https://github.com/williamboman/mason.nvim/blob/f5b1a206c4335e1ccfd4f68feef63ba8950e3e90/lua/mason-registry/rustfmt/init.lua#L17
> > 
> > after remove ` _.gsub("^v", "")`, download works fine, but console print `Link target "/home/jin/.local/share/nvim/mason/packages/rustfmt/rustfmt" does not exist.`
> > I don't know where the downloaded files go？
> > 图片
> 
> I'm having the same issues (both with wrong removal of `v` in the version as well as the linker error)

now i didn't know how to fix it🥹, however i can use rust_analayzer to exec rustfmt fomatting

**Pull request merged**